### DISCUSSION TITLE
LinkTo к временной таблице

### DISCUSSION THREAD

**Discussion by Alexmeb** [totumonline/totum-ru-issues-and-discussions#173] [Как это сделать?] [answered]
Как соединить инсерт во временную и потом ее же открыть как LinkToTable с параметрами field вместо LinkToDataTable?)
Спасибо)

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Логика такая.

Вам надо сначала создать временную таблицу, передав в нее данные для insert, но не показывать ее пользователю, а сделать это в фоне — это делает параметр `hide: true`. Когда он установлен, то вместо открытия таблицы linkToTable возвращает ее hash (временный идентификатор в БД).

А потом открыть ее через linkToTable используя параметр hash:

```
= : linkToTable(table: 'hidden_tmp_table'; hash: $hidden_table; title: "This is a hidden tmp table"; field: 'field_in_tmp' = "some data"; target: "iframe"; width: 800; refresh: true; header: true; footer: true)

hidden_table: linkToDataTable(table: 'hidden_tmp_table'; params: $row; hide: true)

	row: rowCreate(field: "h_insert" = json`[1,2,3,4,5,6,7,8,9,10,11]`)
```

<img width="1306" alt="img" src="https://user-images.githubusercontent.com/55755565/181454823-210c8939-72fe-4b37-b048-f332df974b33.png">
